runtime/common/src/main/kotlin/kotlinx/serialization/modules/SerialModuleBuilders.kt

```diff
@@ -0,0 +1,187 @@
+ * to copy whole another module to this builder with [SerialModule.dumpTo]
+ */
+@Suppress("FunctionName")
+public fun SerializersModule(buildAction: SerializersModuleBuilder.() -> Unit): SerialModule {
+    val builder = SerializersModuleBuilder(SerialModuleImpl())
+    builder.buildAction()
+    return builder.impl
+}
+
+
+/**
+ * A builder inside DSL for building [SerialModule]
+ *
+ * Instance of this builder could be obtained via [SerializersModule] function
+ */
+public class SerializersModuleBuilder internal constructor(internal val impl: SerialModuleImpl) :
+    SerialModuleCollector {
+
+    /**
+     * Add to this module [serializer] which will be used whe [kClass] is
+     * needed to be serialized contextually.
+     *
+     * If serializer already registered for the given KClass, a [SerializerAlreadyRegisteredException] is thrown.
+     * To override registered serializers, combine built module with another using
+     * [SerialModule.overwriteWith].
+     *
+     * @see ContextSerializer
+     */
+    public override fun <T : Any> contextual(kClass: KClass<T>, serializer: KSerializer<T>) =
+        impl.registerSerializer(kClass, serializer)
+
+    /**
+     * Add to this module [actualSerializer] for polymorphic serialization of
+     * [actualClass] in the scope of [baseClass].
+     *
+     * If serializer already registered for the given KClass, a [SerializerAlreadyRegisteredException] is thrown.
+     * To override registered serializers, combine built module with another using
+     * [SerialModule.overwriteWith].
+     */
+    public override fun <Base : Any, Sub : Base> polymorphic(
+        baseClass: KClass<Base>,
+        actualClass: KClass<Sub>,
+        actualSerializer: KSerializer<Sub>
+    ) {
+        impl.registerPolymorphicSerializer(baseClass, actualClass, actualSerializer)
+    }
+
+    /**
+     * Operator-like form of [include].
+     */
+    public operator fun SerialModule.unaryPlus() {
+        this@unaryPlus.dumpTo(this@SerializersModuleBuilder)
+    }
+
+    /**
+     * Copies contents of [other] module into current builder.
+     *
+     * @see SerialModule.unaryPlus
+     */
+    public fun include(other: SerialModule) {
+        other.dumpTo(this)
+    }
+
+    /**
+     * Creates a builder to register all subclasses of a given [baseClass]
+     * for polymorphic serialization. If [baseSerializer] is not null, registers it as a serializer for [baseClass]
+     * (which is useful if base class is serializable). To add subclasses, use
+     * [PolymorphicModuleBuilder.addSubclass] or [PolymorphicModuleBuilder.with].
+     *
+     * If serializer already registered for the given KClass in the given scope, a [SerializerAlreadyRegisteredException] is thrown.
+     * To override registered serializers, combine built module with another using
+     * [SerialModule.overwriteWith].
+     *
+     * @see PolymorphicSerializer
+     */
+    public fun <Base : Any> polymorphic(
```

Comment:
DSL usage shows that we also need `public inline fun <reified Base: Any> polymorphic(baseSerializer: KSerializer<Base>? = null, noinline buildAction: PolymorphicModuleBuilder<Base>.() -> Unit = {}`